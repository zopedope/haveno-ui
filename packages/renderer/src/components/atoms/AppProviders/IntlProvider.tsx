// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
