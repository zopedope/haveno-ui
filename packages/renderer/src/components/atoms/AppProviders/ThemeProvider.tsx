// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
