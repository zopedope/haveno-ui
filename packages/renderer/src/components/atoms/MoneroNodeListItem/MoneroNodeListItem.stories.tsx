// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { Stack } from "@mantine/core";
import type { ComponentStory, ComponentMeta } from "@storybook/react";
import { MoneroNodeListItem, NodeStatus } from ".";

export default {
  title: "atoms/MoneroNodeListItem",
  component: MoneroNodeListItem,
} as ComponentMeta<typeof MoneroNodeListItem>;

const Template: ComponentStory<typeof MoneroNodeListItem> = () => {
  return (
    <Stack>
      <MoneroNodeListItem
        isSelected={true}
        title="node.moneroworldcom:18089"
        status={NodeStatus.Active}
        onClick={() => console.log("clicked")}
      />
      <MoneroNodeListItem
        isSelected={false}
        title="node.xmr.pt:18081"
        status={NodeStatus.Inactive}
        onClick={() => console.log("clicked")}
      />
      <MoneroNodeListItem
        isSelected={false}
        title="node.monero.net:18081"
        status={NodeStatus.Active}
        onClick={() => console.log("clicked")}
      />
    </Stack>
  );
};

export const Default = Template.bind({});
Default.args = {};
