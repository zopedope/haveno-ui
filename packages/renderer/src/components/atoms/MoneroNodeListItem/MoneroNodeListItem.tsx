// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { createStyles, Box, Text, UnstyledButton } from "@mantine/core";

export enum NodeStatus {
  Active = "active",
  Inactive = "inactive",
}

export interface MoneroNodeListItemProps {
  isSelected?: boolean;
  onClick: () => void;
  status: NodeStatus;
  title: string;
}

export function MoneroNodeListItem(props: MoneroNodeListItemProps) {
  const { isSelected = false, onClick, status, title } = props;
  const { classes } = useStyles({ isSelected, status });

  return (
    <UnstyledButton className={classes.root} onClick={onClick}>
      <Text className={classes.title}>{title}</Text>
      <Box className={classes.status}>
        <Box className={classes.statusInner} />
      </Box>
    </UnstyledButton>
  );
}

export const useStyles = createStyles<
  string,
  { isSelected: boolean; status: NodeStatus }
>((theme, { isSelected, status }) => {
  return {
    root: {
      backgroundColor: isSelected
        ? theme.colors.blue[0]
        : theme.colorScheme === "dark"
        ? theme.colors.dark[8]
        : theme.white,
      border: `1px solid ${theme.colors.gray[2]}`,
      borderRadius: theme.radius.md,
      padding: "0.875rem",
      display: "flex",
      transition: "background-color 0.1s ease-in-out",
    },
    title: {
      fontWeight: 600,
      fontSize: theme.fontSizes.sm,
      lineHeight: 1,
      width: "100%",
    },
    status: {
      display: "flex",
    },
    statusInner: {
      height: "0.625rem",
      width: "0.625rem",
      borderRadius: "0.625rem",
      background:
        status === NodeStatus.Active
          ? theme.colors.green[4]
          : theme.colors.gray[4],
      margin: "auto",
    },
  };
});
