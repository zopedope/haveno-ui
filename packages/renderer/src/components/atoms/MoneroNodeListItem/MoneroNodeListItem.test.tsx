// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { MoneroNodeListItem, NodeStatus } from "./MoneroNodeListItem";
import { AppProviders } from "@atoms/AppProviders";

describe("atoms::MoneroNodeListItem", () => {
  it("renders without exploding", () => {
    const { asFragment } = render(
      <AppProviders>
        <MoneroNodeListItem
          isSelected={true}
          title="node.moneroworldcom:18089:active"
          status={NodeStatus.Active}
          onClick={() => console.log("clicked")}
        />
        <MoneroNodeListItem
          isSelected={false}
          title="node.moneroworldcom:18089:inactive"
          status={NodeStatus.Inactive}
          onClick={() => console.log("clicked")}
        />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
