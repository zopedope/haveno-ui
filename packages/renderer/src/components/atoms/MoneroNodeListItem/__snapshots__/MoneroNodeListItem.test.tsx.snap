// Vitest Snapshot v1

exports[`atoms::MoneroNodeListItem > renders without exploding 1`] = `
<DocumentFragment>
  <button
    class="mantine-UnstyledButton-root mantine-146e5sp"
    type="button"
  >
    <div
      class="mantine-Text-root mantine-jmvuvj"
    >
      node.moneroworldcom:18089:active
    </div>
    <div
      class="mantine-1d0mff5"
    >
      <div
        class="mantine-zy87za"
      />
    </div>
  </button>
  <button
    class="mantine-UnstyledButton-root mantine-9tasfa"
    type="button"
  >
    <div
      class="mantine-Text-root mantine-jmvuvj"
    >
      node.moneroworldcom:18089:inactive
    </div>
    <div
      class="mantine-1d0mff5"
    >
      <div
        class="mantine-19v6ci5"
      />
    </div>
  </button>
</DocumentFragment>
`;
